src/coprocessor/codec/mysql/duration.rs

```diff
@@ -13,24 +13,29 @@
 
 use crate::util::codec::number::{self, NumberEncoder};
 use crate::util::codec::BytesSlice;
+use crate::util::escape;
 use std::cmp::Ordering;
 use std::fmt::{self, Display, Formatter};
 use std::io::Write;
 use std::time::Duration as StdDuration;
 use std::{i64, str, u64};
-use time::{self, Tm};
 
-use super::super::Result;
-use super::{check_fsp, parse_frac, Decimal};
+use super::super::{Result, TEN_POW};
+use super::{check_fsp, Decimal};
 
 pub const NANOS_PER_SEC: i64 = 1_000_000_000;
 pub const NANO_WIDTH: u32 = 9;
 const SECS_PER_HOUR: u64 = 3600;
 const SECS_PER_MINUTE: u64 = 60;
+const MAX_HOURS: u64 = 838;
+const MAX_MINUTES: u64 = 59;
+const MAX_SECONDS: u64 = 59;
+const MAX_BLOCK: u64 = 8385959;
```

Comment:
What does MAX_BLOCK mean?

Reply:
> Would you also like to try some existing parser solution like https://github.com/Geal/nom? We usually don't manually write DFAs nowadays when a parser can be used (and is efficient enough) we can use it.

Here is the [parser](https://github.com/mysql/mysql-server/blob/8e797a5d6eb3a87f16498edcb7261a75897babae/mysys/my_time.cc#L597) of MySQL's `TIME` and it's a DFA 😆. BTW, I did try to use `nom`, but it seems the parser we need it's too stateful.

Reply:
> What does MAX_BLOCK mean?

`MAX_BLOCK` means to check the value in the format: `"HHMMSS"`. If the value produced is larger than `MAX_BLOCK`, it's definitely invalid. Otherwise, we will continue the parsing, and check the `MM`/`SS`.

Reply:
Honestly I'm just lazy and want to see how nom performs in this task. Will it as fast as the hand written version, etc 🤣😆

Reply:
Looks like the mysql's implementation is more similar to our current implementation 😆 there is almost no state transitions actually.